### DISCUSSION TITLE
urlencode和变量的嵌套，容易识别错误，如何解决？

### DISCUSSION THREAD

**Discussion by 6Byte** [qd-today/qd#524] [Q&A] [answered]
像这样，有些论坛的用户名必须以GBK编码提交，这就需要编码，像这样 username={{urlencode("{{username}}",encoding="GBK")}}
使用过程中发现这样的嵌套在保存模板的时候公式容易被优化了，导致结果出错？这要如何解决？
![1JPG](https://github.com/qd-today/qd/assets/79741410/37cb9f63-eeaf-443c-944e-470223b80b9f)
![2](https://github.com/qd-today/qd/assets/79741410/cd5e2c3a-f0ea-4de3-b5a8-a60d285a08ed)
![3](https://github.com/qd-today/qd/assets/79741410/4094d2de-f191-47a7-9c10-91fe31e76fe7)

**Comment by 6Byte** [comment]
或者嵌套本来就错误的，应该采用{{username|urlencode}}，但是默认是用utf-8编码，这个形式怎么指定encoding="gbk"呢？

**Comment by a76yyyy** [comment, marked as answer]
{{ username | urlencode(encoding="GBK") }}

**Reply by 6Byte** [reply]
谢谢，这个方法可行！

**Comment by 6Byte** [comment]
谢谢，这个方法可行！
